src/middleware/custom.rs

```diff
@@ -334,65 +336,6 @@ impl CustomPredicateRef {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
```

Comment:
I moved all these lines to `src/middleware/operation.rs` because they are not specific to custom predicate